src/textual/widgets/_selection_list.py

```diff
@@ -0,0 +1,655 @@
+        text-style: bold;
+        background: $foreground 25%;
+    }
+
+    SelectionList > .selection-list--button-selected {
+        text-style: bold;
+        background: $foreground 15%;
+    }
+
+    SelectionList:focus > .selection-list--button-selected {
+        text-style: bold;
+        color: $success;
+        background: $foreground 25%;
+    }
+
+    SelectionList > .selection-list--button-selected-highlighted {
+        text-style: bold;
+        color: $success;
+        background: $foreground 15%;
+    }
+
+    SelectionList:focus > .selection-list--button-selected-highlighted {
+        text-style: bold;
+        color: $success;
+        background: $foreground 25%;
+    }
+    """
+
+    class SelectionMessage(Generic[MessageSelectionType], Message):
+        """Base class for all selection messages."""
+
+        def __init__(self, selection_list: SelectionList, index: int) -> None:
+            """Initialise the selection message.
+
+            Args:
+                selection_list: The selection list that owns the selection.
+                index: The index of the selection that the message relates to.
+            """
+            super().__init__()
+            self.selection_list: SelectionList[MessageSelectionType] = selection_list
+            """The option list that sent the message."""
+            self.selection: Selection[
+                MessageSelectionType
+            ] = selection_list.get_option_at_index(index)
+            """The highlighted selection."""
+            self.selection_index: int = index
+            """The index of the selection that the message relates to."""
+
+        @property
+        def control(self) -> OptionList:
+            """The selection list that sent the message.
+
+            This is an alias for
+            [`SelectionMessage.selection_list`][textual.widgets.SelectionList.SelectionMessage.selection_list]
+            and is used by the [`on`][textual.on] decorator.
+            """
+            return self.selection_list
+
+        def __rich_repr__(self) -> Result:
+            yield "selection_list", self.selection_list
+            yield "selection", self.selection
+            yield "selection_index", self.selection_index
+
+    class SelectionHighlighted(SelectionMessage):
+        """Message sent when a selection is highlighted.
+
+        Can be handled using `on_selection_list_selection_highlighted` in a subclass of
+        [`SelectionList`][textual.widgets.SelectionList] or in a parent node in the DOM.
+        """
+
+    class SelectionToggled(SelectionMessage):
+        """Message sent when a selection is toggled.
+
+        Can be handled using `on_selection_list_selection_toggled` in a subclass of
+        [`SelectionList`][textual.widgets.SelectionList] or in a parent node in the DOM.
+
+        Note:
+            This message is only sent if the selection is toggled by user
+            interaction. See
```

Comment:
Why?

Reply:
Because the selection-oriented messages are really just thin wraps/replacements of the user-interaction-oriented messages in `OptionList`. With `SelectionList` the thing the developer really wants to care about is when the selection list as a whole had changed (`SelectedChanged`).